vividus-extension-selenium/src/main/java/org/vividus/steps/ui/GenericWaitSteps.java

```diff
@@ -269,6 +273,36 @@ public void doesElementByLocatorExistsForDuration(Locator locator, Duration dura
         });
     }
 
+    /**
+     * Waits until an element with the specified locator has text that matches the provided regular expression.
+     *
+     * @param locator The locator of the element whose text to check
+     * @param regex   The regular expression used to validate the text of the element
+     */
+    @When("I wait until element located by `$locator` has text matching `$regex`")
+    public void waitUntilElementHasTextMatchingRegex(Locator locator, Pattern regex)
+    {
+        uiContext.getOptionalSearchContext()
+                .ifPresent(context -> waitActions.wait(context, new IExpectedSearchContextCondition<>()
+                {
+                    @Override
+                    public Boolean apply(SearchContext searchContext)
+                    {
+                        return searchActions.findElement(locator).map(elementActions::getElementText)
+                                                                 .map(regex::matcher)
+                                                                 .map(java.util.regex.Matcher::find)
+                                                                 .orElse(false);
+                    }
+
+                    @Override
+                    public String toString()
+                    {
+                        return String.format("text matching regex ('%s') to be present in element located by %s", regex,
+                                locator);
```

Comment:
how does passed and failed steps look like?

Reply:
Failed:
<img width="1664" alt="image" src="https://github.com/vividus-framework/vividus/assets/30923336/ab9575e7-cc23-4c27-82f7-22349337bf28">

Reply:
shared privately